inc/shortcodes/class-shortcode.php

```diff
@@ -80,6 +80,28 @@ protected static function parse_url( $url, $component = -1 ) {
 	}
 
 	/**
+	 * Parse a string of content for any oembeds that can be potentially reversed.
+	 *
+	 * @param string $content
+	 * @return array|false An array with the full lines matched as keys, and the URLs as values
+	 */
+	protected static function parse_oembeds( $content ) {
+		if ( false === stripos( $content, 'http' ) ) {
+			return false;
+		}
+
+		if ( preg_match_all( '#^(\s*)(https?://[^\s"]+)(\s*)$#im', $content, $matches, PREG_SET_ORDER ) ) {
```

Comment:
This is going to be pretty slow to run so frequently, particularly because it's matched with more regex.